lib/scenario_runner.py

```diff
@@ -1304,6 +1314,48 @@ def _setup_services(self):
                 'read-sci-stdout': service.get('read-sci-stdout', False),
             }
 
+            # Check if detached container failed immediately after startup
+            # Docker run -d returns exit code 0 (success) even when containers fail moments later
+            # Common causes: architecture mismatch, missing dependencies, invalid entrypoints
+            time.sleep(1)
+            check_ps = subprocess.run(
+                    ['docker', 'ps', '-q', '-f', f'name={container_name}'],
+                    capture_output=True,
+                    text=True,
+                    check=False
+                )
+            if not check_ps.stdout.strip():
+                # Container not running anymore!
+                logs_ps = subprocess.run(
+                    ['docker', 'logs', container_name],
+                    capture_output=True,
+                    text=True,
+                    check=False
+                )
+                inspect_ps = subprocess.run(
+                    ['docker', 'inspect', '--format={{.State.ExitCode}}', container_name],
+                    capture_output=True,
+                    text=True,
+                    check=False
+                )
+                exit_code = inspect_ps.stdout.strip() if inspect_ps.returncode == 0 else "unknown"
+
+                if exit_code == "0":
+                    # Container exited successfully but immediately - unexpected behavior, log but continue
```

Comment:
Since the container has been added to the __containers  list just logging will not suffice. GMT will fail later as it cannot attach reporters. I would except here.

Reply:
Changed it so it will raise an exception. However, this has one disadvantage: Test cases with containers that exit intermediately are not possible anymore. See test case `test_cmd_entrypoint`. I did not want to extend the duration of this test case. Therefore I introduced a new argument of the ScenarioRunner: `enable_successful_container_start_check`

Not sure, if this is a good solution. Alternative would be to extend the duration of the test case.